File: python_code/ContainsDuplicate.py
class ContainsDuplicate:
    @staticmethod
    def contains_duplicate(nums):
        seen = set()  # ①

        for num in nums:  # ②
            if num in seen:  # ③
                return True  # ④
            seen.add(num)  # ⑤

        return False  # ⑥

# Ejemplo de uso:
nums = [1, 2, 3, 1]
has_duplicate = ContainsDuplicate.contains_duplicate(nums)
print("¿Contiene duplicados?:", has_duplicate)  # Resultado esperado: True

File: ts_code/ContainsDuplicate.ts
class ContainsDuplicate {
    public static containsDuplicate(nums: number[]): boolean {
        const seen = new Set<number>(); // ①

        for (const num of nums) { // ②
            if (seen.has(num)) {    // ③
                return true;          // ④
            }
            seen.add(num);          // ⑤
        }

        return false; // ⑥
    }
}

// Ejemplo de uso:
const nums_: number[] = [1, 2, 3, 1];
const hasDuplicate = ContainsDuplicate.containsDuplicate(nums_);
console.log("¿Contiene duplicados?:", hasDuplicate); // Resultado esperado: true
